lib/modules/hosts/tenor.js

```diff
@@ -0,0 +1,64 @@
+import { ajax } from '../../environment';
+
+
+const _alphabet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'.split('');
+const _alphabetMap = {};
+_alphabet.forEach((c, i) => {
+	_alphabetMap[c] = i;
+});
+
+const tenor = {
+	moduleID: 'tenor',
+	name: 'tenor',
+	domains: ['tenor.co'],
+	logo: 'https://www.tenor.co/favicon.ico',
+	parseViewShortId: s => {
+		s = s.split('');
+		let n = 0;
+
+		s.forEach(c => {
+			n = (n * _alphabet.length) + _alphabetMap[c];
+		});
```

Comment:
1. We have short URLs in the form "https://tenor.co/s2Vn.gif", where "s2Vn" is en encoded int. This function decodes that int ID.
2. Just rewrote it to use reduce.